pandas/core/internals/managers.py

```diff
@@ -1988,9 +1993,34 @@ def setitem_inplace(self, indexer, value) -> None:
         in place, not returning a new Manager (and Block), and thus never changing
         the dtype.
         """
-        if using_copy_on_write() and not self._has_no_reference(0):
-            self.blocks = (self._block.copy(),)
-            self._cache.clear()
+        if not self._has_no_reference(0):
```

Comment:
I'd rather not check references before establishing that we are in the correct mode after our performance issues

Reply:
Good point, will update